core/reference.cpp

```diff
@@ -42,6 +42,8 @@ bool Reference::init_ref() {
 		if (refcount_init.get() > 0) {
 			refcount_init.unref();
 			refcount.unref(); // first referencing is already 1, so compensate for the ref above
+		} else if (get_script_instance()) {
```

Comment:
This is needed so any script that wants to use the functionality gets notified about refcount changes. This is particularly useful for GDNative when used in languages with integrated GC.

Reply:
OK, wouldn't it make more sense to make void Reference::reference() a return a bool, like SafeRefCount::ref(), then call that instead of duplicating the reference() functionality here?

Reply:
Changed init_ref to use reference()/unreference().